atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -112,6 +114,10 @@ public Properties getHikariProperties() {
 
         props.setProperty("oracle.jdbc.maxCachedBufferSize", "100000");
 
+        if (getDefaultRowPrefetch().isPresent()) {
+            props.setProperty("defaultRowPrefetch", getDefaultRowPrefetch().get().toString());
```

Comment:
Can we add tests confirming this is taking effect properly? One potential concern here is that setting this too high increases likelihood of OOMing client, so should definitely document this and provide some recommendations on when to increase as this will increase it for *all* result sets, including those outside the scope of Atlas if they share this connection pool. Note that for Atlas tables, the memory footprint overhead is fairly straightforward to calculate, so we might want to consider explicitly setting the fetch size on the Atlas specific `*Statement`s and `ResultSet`s themselves.

Would be good to use consistent configuration names across both Postgres & Oracle, maybe bikeshedding, but how about `defaultResultSetFetchSize`? Note that we should document that if not set, we'll use the default of the underlying DB (Postgres default is 0 which means fetch all rows, Oracle default is 10 rows per fetch). 

Probably worth a comment linking https://docs.oracle.com/cd/E18283_01/appdev.112/e13995/constant-values.html#oracle_jdbc_OracleConnection_CONNECTION_PROPERTY_DEFAULT_ROW_PREFETCH as well

Might also want to consider allowing specification of `"useFetchSizeWithLongColumn"` per https://docs.oracle.com/cd/E11882_01/java.112/e16548/oraperf.htm#JJDBC28781 though more generally, might be worth allowing pass-through of arbitrary property mappings to the underlying JDBC driver.